valohai_yaml/pipelines/conversion.py

```diff
@@ -52,6 +53,38 @@ def convert_deployment_node(self, node: DeploymentNode) -> ConvertedObject:
         node_data["aliases"] = node.aliases
         return node_data
 
+    def _merge_overridden_inputs(self, overridden: List[OrderedDict], list2: List[OrderedDict]) -> List[OrderedDict]:
+        merged_list = []
+        dict1_map = {ordered_dict['name']: ordered_dict for ordered_dict in overridden}
+        for ordered_dict2 in list2:
+            name = ordered_dict2['name']
+            if name in dict1_map:
+
+                ordered_dict1 = dict1_map[name]
+                merged_dict = OrderedDict()
+
+                for key, value in ordered_dict1.items():
+                    if key == 'default':
+                        merged_dict[key] = self._merge_defaults(value, ordered_dict2.get(key))
+                    else:
+                        merged_dict[key] = value
+
+                merged_list.append(merged_dict)
+            else:
+                merged_list.append(ordered_dict2)
+
+        for ordered_dict1 in overridden:
+            name = ordered_dict1['name']
+            if name not in dict1_map:
+                merged_list.append(ordered_dict1)
+
+        return merged_list
+
+    def _merge_defaults(self, default1: Optional[Any], default2: Optional[Any]) -> List[str]:
+        default1 = [default1] if isinstance(default1, str) else (default1 if default1 else [])
+        default2 = [default2] if isinstance(default2, str) else (default2 if default2 else [])
```

Comment:
I think these two `merge` functions could be made free functions and maybe moved to `pipelines/overrides.py` or something?